elastic/kibana
# [SecuritySolution] Track ui metrics for landing pages, nav, and breadcrumbs

## Summary
Original issue:
https://github.com/elastic/kibana/issues/144945

Dashboard (staging): 
https://telemetry-v2-staging.elastic.dev/s/securitysolution/app/dashboards#/view/257918f0-b450-11ed-a6e6-d32d2209b7b7?_g=()

Steps to verify:
1. Click on navigations, landing page items, and bread crumbs.
2. Create some dashboards from `dashboard` landing page.
3. Visit http://localhost:5601/api/stats?extended=true&legacy=true, check `ui_counters` for the clicked events. Find `security_solution dashboardMetics` for dashboard usage.

This PR implements:
1. ui tracking(click) for navigations, landing pages, bread crumbs
2. The usage of custom dashboards - the dashboard saved objects labeled with `security solution`
3. ui tracking `legacy_navigation_` for old navigations

<img width="2541" alt="navigation_ui_metric-1" src="https://user-images.githubusercontent.com/6295984/217849293-620c7038-44f0-4a2f-888e-f1324df6f01c.png">


<img width="834" alt="breadcrumps_ui_metric" src="https://user-images.githubusercontent.com/6295984/217849088-818fb030-7156-4af3-b483-473b7b8c3494.png">

<img width="907" alt="grouped_nav_manage" src="https://user-images.githubusercontent.com/6295984/217849104-dd3caa6d-436a-416e-b650-ce3acfb90828.png">
<img width="2549" alt="landing_page_menege_ui-Metrics" src="https://user-images.githubusercontent.com/6295984/217853256-a7d63cb3-791f-46b0-92cc-fe9f213d53b9.png">


<img width="1515" alt="grouped_nav_explore" src="https://user-images.githubusercontent.com/6295984/217849187-ca769f0a-5e53-403b-bda5-111ddd82d8ac.png">
<img width="591" alt="grouped_nav_dashboards" src="https://user-images.githubusercontent.com/6295984/217849259-248b8b42-1ac6-45ea-9558-bbcf7f2a60ff.png">
<img width="2544" alt="navigation_ui_metric-2" src="https://user-images.githubusercontent.com/6295984/217849275-bdc86312-ae29-4de6-af54-e9a3e6cc0e23.png">
<img width="2545" alt="landing_page_explore_ui_metrics" src="https://user-images.githubusercontent.com/6295984/217849298-1987286e-1a8b-451e-b7f4-421427c6412b.png">
<img width="2543" alt="landing_page_dashboards_ui_metrics" src="https://user-images.githubusercontent.com/6295984/217848984-56616698-0cc7-498c-b116-d6f22c823ece.png">


### Checklist

Delete any items that are not applicable to this PR.
